jdk/src/share/classes/java/util/Formatter.java

```diff
@@ -4380,9 +4380,11 @@ private char getZero(Locale l) {
 
             // apply zero padding
             len = sb.length();
-            if (width != -1 && f.contains(Flags.ZERO_PAD))
-                for (int k = 0; k < width - len; k++)
-                    sb.insert(begin, zero);
+            if (width > len && f.contains(Flags.ZERO_PAD)) {
+                char[] zeros = new char[width - len];
+                Arrays.fill(zeros, zero);
+                sb.insert(begin, new String(zeros));
```

Comment:
Oh wait, we actually have `StringBuilder.append(char[])`! That's useful, we don't need to do `new String()` then?